io-engine/src/core/volume_snapshot.rs

```diff
@@ -0,0 +1,36 @@
+use super::CoreError;
+use crate::subsys::NvmfReq;
+use async_trait::async_trait;
+use spdk_rs::libspdk::spdk_bdev_io;
+
+///  Trait gives the interface for all Snapshot and Clone related interfaces
+#[async_trait(?Send)]
+pub trait VolumeSnapshot {
+    type Snapshot;
+
+    /// Create Snapshot (Readonly Copy) for Local Volume
+    async fn create_snapshot_local(
+        &self,
+        snap_name: &str,
+        bdev_req: *mut spdk_bdev_io,
+    ) -> Result<u64, CoreError>;
+
+    /// Create Snapshot (Readonly Copy) for Remote Volume
+    async fn create_snapshot_remote(
+        &self,
+        snap_name: &str,
+        nvmf_req: &NvmfReq,
+    ) -> Result<u64, CoreError>;
+
+    /// Return the list of Snapshots
+    fn list_snapshot(&self, source_vol: &str) -> Vec<Self::Snapshot>;
```

Comment:
What's the semantics of this method ? List snapshots for a snapshot ?

Reply:
It means, to list the snapshots of of a particular source volume, from which multiple snapshots are created.

Reply:
I think that list_snapshot should go under the volume trait